graviti/portex/field.py

```diff
@@ -85,6 +87,190 @@ def rename(self, old_name: str, new_name: str) -> None:
         self._keys.__setitem__(self._keys.index(old_name), new_name)
 
 
+class Indexes:
+    """The indexes manager of the connected fields.
+
+    Arguments:
+        total_count: The total count of the elements in the paging list.
+        limit: The size of each page.
+
+    """
+
+    def __init__(self, lengths: Iterable[int]) -> None:
+        self._indexes = [0, *accumulate(lengths)]
+        print("indexes", self._indexes)
+
+    def __getitem__(self, index: int) -> int:
+        return self._indexes[index]
+
+    def get_index(self, index: int) -> Tuple[int, int]:
+        """Get the index of the field.
+
+        Arguments:
+            index: The index of the field in connected fields.
+
+        Returns:
+            The fields number and the index of the field.
+
+        """
+        i = bisect_right(self._indexes, index) - 1
+        return i, index - self._indexes[i]
+
+    def update(self, index: int, length: int = 1) -> None:
+        """Update the indexes with index and length changes.
+
+        Arguments:
+            index: The index where the changes happened.
+            length: The length change of the index.
+
+        """
+        for i in range(index + 1, len(self._indexes)):
+            self._indexes[i] += length
+        print("updated indexes", self._indexes)
+
+
+class ConnectedFields:
```

Comment:
why not set the base class of ConnectedFields to `NameOrderedDict` or `MutableMapping`?